Model/Push/PushProcessorsFactory.php

```diff
@@ -0,0 +1,115 @@
+
+    /**
+     * @var array
+     */
+    protected array $pushProcessors;
+
+    /**
+     * @var ?PushProcessorInterface
+     */
+    protected ?PushProcessorInterface $pushProcessor = null;
+
+    /**
+     * @var OrderRequestService
+     */
+    private OrderRequestService $orderRequestService;
+
+    /**
+     * @param OrderRequestService $orderRequestService
+     * @param ObjectManagerInterface $objectManager
+     * @param array $pushProcessors
+     */
+    public function __construct(
+        OrderRequestService $orderRequestService,
+        ObjectManagerInterface $objectManager,
+        array $pushProcessors = []
+    ) {
+        $this->objectManager = $objectManager;
+        $this->pushProcessors = $pushProcessors;
+        $this->orderRequestService = $orderRequestService;
+    }
+
+    /**
+     * Retrieve proper push processor for the specified transaction method.
+     *
+     * @param PushTransactionType|null $pushTransactionType
+     * @return ?PushProcessorInterface
+     * @throws BuckarooException
+     */
+    public function get(?PushTransactionType $pushTransactionType): ?PushProcessorInterface
+    {
+        if (!$this->pushProcessor instanceof PushProcessorInterface) {
+            if (empty($this->pushProcessors)) {
+                throw new \LogicException('Push processors is not set.');
+            }
+
+            $pushProcessorClass = $this->getPushProcessorClass($pushTransactionType);
+            if (empty($pushProcessorClass)) {
+                throw new BuckarooException(new Phrase('Unknown Push Processor type'));
+            }
+
+            $this->pushProcessor = $this->objectManager->get($pushProcessorClass);
+
+        }
+        return $this->pushProcessor;
+    }
+
+    /**
+     * @param PushTransactionType|null $pushTransactionType
+     * @return mixed
+     * @throws BuckarooException
+     */
+    private function getPushProcessorClass(?PushTransactionType $pushTransactionType)
+    {
+        // Set Default Push Processor
+        $pushProcessorClass = $this->pushProcessors['default'];
+
+        // Set Push Processor by Payment Method
+        $paymentMethod = $pushTransactionType->getPaymentMethod();
+        $pushProcessorClass = $this->pushProcessors[$paymentMethod] ?? $pushProcessorClass;
+        if ($pushTransactionType->isFromPayPerEmail()) {
+            $this->pushProcessors['group_transaction'];
+        }
+
+        // Check if is Group Transaction Push
+        if ($pushTransactionType->isGroupTransaction()) {
+            $pushProcessorClass = $this->pushProcessors['group_transaction'];
+        }
+
+        // Check if is Credit Management Push
+        $pushType = $pushTransactionType->getPushType();
+        if ($pushType == PushTransactionType::BUCK_PUSH_TYPE_INVOICE) {
+            $pushProcessorClass = $this->pushProcessors['credit_managment'];
+        } elseif ($pushType == PushTransactionType::BUCK_PUSH_TYPE_INVOICE_INCOMPLETE) {
```

Comment:
in my opinion we should do simple ifs and return, will look cleaner

Reply:
I proceed like this because a pushProcessorClass can change the value from $paymentMethod to group_transaction or refund. Also I proceed like this to avoid the next error: "This method has 5 returns, which is more than the 3 allowed."